core/src/test/java/kafka/admin/UserScramCredentialsCommandTest.java

```diff
@@ -0,0 +1,178 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.admin;
+
+import kafka.server.BaseRequestTest;
+import kafka.utils.Exit;
+import org.apache.kafka.test.TestUtils;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Console;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.OptionalInt;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings("dontUseSystemExit")
+public class UserScramCredentialsCommandTest extends BaseRequestTest {
+    private static final Logger log = LoggerFactory.getLogger(UserScramCredentialsCommandTest.class);
+
+    @Override
+    public int brokerCount() {
+        return 1;
+    }
+
+    static class ConfigCommandResult {
+        public final String stdout;
+        public final OptionalInt exitStatus;
+
+        public ConfigCommandResult(String stdout) {
+            this(stdout, OptionalInt.empty());
+        }
+
+        public ConfigCommandResult(String stdout, OptionalInt exitStatus) {
+            this.stdout = stdout;
+            this.exitStatus = exitStatus;
+        }
+    }
+
+    private ConfigCommandResult runConfigCommandViaBroker(String...args) throws UnsupportedEncodingException {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        String utf8 = StandardCharsets.UTF_8.name();
+        PrintStream printStream = new PrintStream(byteArrayOutputStream, true, utf8);
+        AtomicReference<OptionalInt> exitStatus = new AtomicReference<>(OptionalInt.empty());
+        Exit.setExitProcedure((status, __) -> {
+            exitStatus.set(OptionalInt.of((Integer) status));
+            throw new RuntimeException();
+        });
+
+        List<String> commandArgs = new ArrayList<>(Arrays.asList("--bootstrap-server", bootstrapServers(listenerName())));
+        commandArgs.addAll(Arrays.asList(args));
+        try {
+            Console.withOut(printStream, () -> {
+                ConfigCommand.main(commandArgs.toArray(new String[0]));
+                return null;
+            });
+            return new ConfigCommandResult(byteArrayOutputStream.toString(utf8));
+        } catch (Exception e) {
+            log.debug("Exception running ConfigCommand " + String.join(" ", commandArgs), e);
```

Comment:
Could you please remove this debug message? We can add it back if it gets flaky

Reply:
yes. removed